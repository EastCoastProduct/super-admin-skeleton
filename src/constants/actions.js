export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';

export const USERS_GET_FAILED = 'USERS_GET_FAILED';
export const USERS_GET_SUCCESS = 'USERS_GET_SUCCESS';
export const USERS_PAGINATION_CHANGE = 'USERS_PAGINATION_CHANGE';

export const USER_GET_FAILED = 'USER_GET_FAILED';
export const USER_FETCH_SUCCESS = 'USER_FETCH_SUCCESS';
