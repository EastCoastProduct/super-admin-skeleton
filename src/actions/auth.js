import store from 'store';
import { LOGIN_SUCCESS, LOGOUT_SUCCESS } from '../constants/actions';
import { API_URL } from '../constants/application';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const loginSuccess = superadmin => ({
  type: LOGIN_SUCCESS,
  superadmin,
});

export const logoutSuccess = () => ({
  type: LOGOUT_SUCCESS,
});

export const loginFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/authenticate`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      store.set('token', `Bearer ${resp.token}`);
      store.set('superadmin', resp.user);
      dispatch(loginSuccess(resp.user));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const logoutAction = cb =>
  (dispatch) => {
    store.clear();
    dispatch(logoutSuccess());
    return typeof cb === 'function' && cb();
  };
