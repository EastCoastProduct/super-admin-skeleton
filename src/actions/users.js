import queryString from 'query-string';
import { USERS_GET_SUCCESS, USERS_GET_FAILED, USERS_PAGINATION_CHANGE }
  from '../constants/actions';
import { API_URL } from '../constants/application';
import fetch from '../utils/fetch';

export const usersGetSuccess = (list, listTotal) => ({
  type: USERS_GET_SUCCESS,
  list,
  listTotal,
});

export const usersGetFailed = error => ({
  type: USERS_GET_FAILED,
  error,
});

export const paginationChange = page => ({
  type: USERS_PAGINATION_CHANGE,
  page,
});

export const usersGetFetch = (qs, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users?${queryString.stringify(qs)}`)
      .then((resp) => {
        dispatch(usersGetSuccess(resp.rows, resp.count));
        return typeof cb === 'function' && cb();
      }).catch(err =>
        dispatch(usersGetFailed(err.message))
      );
