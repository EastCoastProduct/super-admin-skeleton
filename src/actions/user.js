import { USER_FETCH_SUCCESS, USER_GET_FAILED } from '../constants/actions';
import { API_URL } from '../constants/application';
import createFormData from '../utils/createFormData';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const userFetchSuccess = profile => ({
  type: USER_FETCH_SUCCESS,
  profile,
});

export const userGetFailed = error => ({
  type: USER_GET_FAILED,
  error,
});

export const userGetFetch = userId =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`).then(resp =>
      dispatch(userFetchSuccess(resp))
    ).catch(err =>
      dispatch(userGetFailed(err.message))
    );

export const userUpdateFetch = (values, userId, cb) =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`, {
      method: 'POST',
      body: createFormData(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const userCreateFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb(resp.id);
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );
