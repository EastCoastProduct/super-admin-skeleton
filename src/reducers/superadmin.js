import store from 'store';
import { fromJS } from 'immutable';
import { LOGIN_SUCCESS } from '../constants/actions';

const initialState = () => fromJS(store.get('superadmin')) || fromJS({});

export default (state = initialState(), action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      return state.merge(fromJS(action.superadmin));
    default:
      return state;
  }
};
