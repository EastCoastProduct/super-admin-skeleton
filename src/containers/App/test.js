import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { superadmin } from '../../fixtures/superadmin';
import { AppComponent } from './';
import * as Actions from '../../actions/auth';

describe('App component', () => {
  const props = {
    dispatch: jest.fn(),
    router: {
      push: jest.fn(),
    },
    superadmin: fromJS(superadmin),
  };
  const wrapper = shallow(
    <AppComponent {...props}>
      <main>Test</main>
    </AppComponent>
  );

  describe('snapshot', () => {
    it('renders basic required data', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('instance', () => {
    afterEach(() => {
      jest.resetAllMocks(); // move to before when clearAllMocks gets out
    });
    Actions.logoutAction = jest.fn();
    const instance = wrapper.instance();

    it('handleLogout method', () => {
      const event = {
        preventDefault: jest.fn(),
      };
      instance.handleLogout(event);

      expect(event.preventDefault).toHaveBeenCalledTimes(1);
      expect(Actions.logoutAction).toHaveBeenCalled();
      expect(instance.props.dispatch).toHaveBeenCalled();
      expect(instance.props.router.push).toHaveBeenCalledWith('/login');
    });
  });
});
