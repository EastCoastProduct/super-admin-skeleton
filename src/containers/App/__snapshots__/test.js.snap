exports[`App component snapshot renders basic required data 1`] = `
<div>
  <Header
    handleLogout={[Function]}
    superadmin={
      Object {
        "bio": null,
        "createdAt": "2016-11-24T14:48:32.000Z",
        "email": "super.admin@mail.com",
        "firstname": "superadmin",
        "id": 1,
        "lastname": "superadmin",
        "updatedAt": "2016-11-24T14:48:32.000Z",
      }
    } />
  <Navigation />
  <div
    className="content_dqivkm">
    <main>
      Test
    </main>
  </div>
</div>
`;
